Generic enums with concrete type aliases

nightscape opened the pull request:
## Changes

Fixes #2252 

The idea of this PR is to be able to do something like this:

```rust
pub enum Change<T> {
    Created { data: T },
    Updated { id: String, data: T },
    Deleted { id: String },
}

pub type StringChange = Change<String>;
pub type MapChange = Change<HashMap<String, String>>;
```

## Procedure and Checklist

In order to quickly iterate and avoid slowing down development pace by making CI pass, only the following simplified steps are needed, and I (fzyzcjy) will handle the rest of CI / moving the tests currently (will have more automation in the future).

- [x] Implement the feature / fix the bug.
- [x] Add tests in `frb_example/dart_minimal`.
- [x] Make `dart_minimal`'s CI green.

Utility commands
- Run codegen: `cargo run --manifest-path /path/to/flutter_rust_bridge/frb_codegen/Cargo.toml -- generate`
- Run tests: `./frb_internal test-dart-native --package frb_example/dart_minimal`

nightscape committed:
Generic enums with concrete type aliases

fzyzcjy commented:
Great job! Feel free to ping me if need any suggestions

nightscape commented:
@fzyzcjy I've made a few more fixes and it works reliably now for my use case.
Would you mind having a look?

fzyzcjy commented on frb_example/pure_dart/rust/src/api/enumeration.rs:
```diff
@@ -57,6 +59,80 @@ pub fn func_enum_with_discriminant_twin_normal(
+}
+
+#[frb(non_opaque)]
+pub struct Butch<T> {
```
maybe AnotherBatchTwiNormal

nightscape replied:
Renamed to `AnotherBatchTwinNormal`

fzyzcjy commented on frb_example/pure_dart/rust/src/api/enumeration.rs:
Before digging into details, I wonder whether it will generalize to generic structs (seems you add some fields to struct logic, but there are no tests yet)

fzyzcjy reviewed (commented):
Excellent job!

nightscape commented:
@fzyzcjy @rlch I
- added generic struct handling
- rebased the branch
- incorporated the review comment from Gemini

I have no idea about the test failures. Seems the git checkout of `cargokit` failed for some reason.

nightscape commented on frb_example/pure_dart/rust/src/api/enumeration.rs:
```diff
@@ -57,6 +59,80 @@ pub fn func_enum_with_discriminant_twin_normal(
+    arg
+}
+
+pub struct Batch<T> {
```
Renamed to `BatchTwinNormal`

fzyzcjy commented:
No worries! I plan to have a try to handle all prs w/ the help of llm agents to fix things and merge this (and other) prs.
